src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,60 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationServiceException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkValidLogin(user);
+        checkPassword(user);
+        checkAge(user);
+        return storageDao.add(user);
+    }
+
+    private void checkValidLogin(User user) {
+        if (user.getLogin() == null
+                || user.getLogin().length() < MIN_LENGTH) {
+            throw new RegistrationServiceException(
+                    "User login can not be least then "
+                            + MIN_LENGTH
+                            + " chars!"
+            );
```

Comment:
seperate the null check from the min length check, because in case you pass a null login, the exception will tell us about the length problem and not the null, which is misguiding.
Same for other validations

Reply:
fixed